File: xchange/src/main/java/com/cattle/xchange/domain/cattle/cattleAd/dtos/CattleAdForBuyMinDTO.java
package com.cattle.xchange.domain.cattle.cattleAd.dtos;

import com.cattle.xchange.domain.cattle.cattleAd.CattleAd;
import jakarta.validation.constraints.NotBlank;

import java.util.UUID;

public record CattleAdForBuyMinDTO
        (@NotBlank
         String title,
         @NotBlank
         double unitValue,
         @NotBlank
         int quantity,
         @NotBlank
         UUID userCod
        ) {
    public CattleAdForBuyMinDTO(CattleAd user) {
        this(
                user.getTitle(),
                user.getUnitValue(),
                user.getQuantity(),
                user.getUserCod()
        );
    }
}

File: Front/cattle-xchange/src/app/model/cattleAd/cattle/dtos/cattle-ad-for-buy-min-dto.ts
import { CattleAdLink } from '../cattle-ad-link';

export interface CattleAdForBuyMinDTO {
  id: string;
  title: string;
  unitValue: number;
  quantity: number;
  ownerCod: string;
  cattleAdLink: CattleAdLink;
}
